loadtest/control/gencontroller/actions.go

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package gencontroller
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/store"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/store/memstore"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type userAction struct {
+	run        control.UserAction
+	frequency  int
+	idleTimeMs int
+}
+
+func logout(u user.User) control.UserActionResponse {
+	ok, err := u.Logout()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if !ok {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("user did not logout"))}
+	}
+
+	return control.UserActionResponse{Info: "logged out"}
+}
+
+func (c *GenController) createTeam(u user.User) control.UserActionResponse {
+	if !st.inc("teams", c.config.NumTeams) {
+		return control.UserActionResponse{Info: "target number of teams reached"}
+	}
+
+	team := &model.Team{
+		AllowOpenInvite: true,
+		Type:            "O",
+	}
+	team.Name = "team-" + model.NewId()
```

Comment:
Why we are adding an ID here? To be able to get a random name?

Reply:
Yes. The reason I am not calling just `model.NewId()` is because there are some disallowed prefixes for the name of a team like `api` which could get randomly generated and would cause a failure.